test/test_other.py

```diff
@@ -12065,6 +12065,18 @@ def test_pthread_hello(self, args):
   def test_pthread_relocatable(self):
     self.do_run_in_out_file_test('hello_world.c', emcc_args=['-sRELOCATABLE'])
 
+  @node_pthreads
+  def test_pthread_unavailable(self):
+    # Run a simple hello world program that uses pthreads
+    self.emcc_args += ['-sPROXY_TO_PTHREAD', '-sEXIT_RUNTIME']
+    self.do_run_in_out_file_test('hello_world.c')
+
+    # Now run the same program but with SharedArrayBuffer undefined, it should run
+    # fine and then fail on the first call to pthread_create.
+    create_file('pre.js', 'SharedArrayBuffer = undefined\n')
+    expected = 'pthread_create: environment does not support SharedArrayBuffer, pthreads are not available'
+    self.do_runf('hello_world.c', expected, assert_returncode=NON_ZERO, emcc_args=['--pre-js=pre.js'])
```

Comment:
Maybe it's worth adding a specific test here that verifies we can do things up until the pthread creation? Like a program with main
```c
int main() {
  printf("This is printed\n");
  pthread_create(..)
  printf("This is not printed\");
}
```
Then we can verify we error at the `pthread_create`, but managed to print the first line.

Reply:
I think that fact that the error message starts with `pthread_create:` is enough to verify this.